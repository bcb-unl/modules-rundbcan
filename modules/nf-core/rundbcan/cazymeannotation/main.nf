process RUNDBCAN_CAZYMEANNOTATION {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/dbcan:5.0.4--pyhdfd78af_0' :
        'biocontainers/dbcan:5.0.4--pyhdfd78af_0' }"
    input:
    tuple val(meta), path(input_raw_data)
    path dbcan_db

    output:
    tuple val(meta), path("${prefix}/overview.tsv")            , emit: cazyme_annotation
    tuple val(meta), path("${prefix}/dbCAN_hmm_results.tsv")   , emit: dbcanhmm_results
    tuple val(meta), path("${prefix}/dbCANsub_hmm_results.tsv"), emit: dbcansub_results
    tuple val(meta), path("${prefix}/diamond.out")             , emit: dbcandiamond_results

    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_dbcan_cazyme"
    def VERSION = '5.0.4'

    """

    run_dbcan CAZyme_annotation \\
        --mode protein \\
        --db_dir ${dbcan_db} \\
        --input_raw_data ${input_raw_data} \\
        --output_dir ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_dbcan_cazyme"
    def VERSION = '5.0.4'
    """
    mkdir -p ${prefix}
    touch ${prefix}/overview.tsv
    touch ${prefix}/dbCAN_hmm_results.tsv
    touch ${prefix}/dbCANsub_hmm_results.tsv
    touch ${prefix}/diamond.out

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """
}
