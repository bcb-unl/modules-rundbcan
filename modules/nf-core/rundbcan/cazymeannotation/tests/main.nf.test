nextflow_process {

    name "Test Process RUNDBCAN_CAZYMEANNOTATION"
    script "../main.nf"
    process "RUNDBCAN_CAZYMEANNOTATION"

    tag "modules"
    tag "modules_nfcore"
    tag "rundbcan"
    tag "rundbcan/database"
    tag "rundbcan/cazymeannotation"

    test("dbcancazyme - simplified") {

        setup {
            run("RUNDBCAN_DATABASE"){
                script "../../database/main.nf"
                process {
                    """
                    """
                }

            }
        }


        when {
            process {
                """
                input[0] = [
                    [id:'test_dbcan'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/proteome.fasta', checkIfExists: true)
                ]
                input[1] = RUNDBCAN_DATABASE.out.dbcan_db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert path(process.out.cazyme_annotation.get(0).get(1)).exists() },
                { assert path(process.out.dbcanhmm_results.get(0).get(1)).exists() },
                { assert path(process.out.dbcansub_results.get(0).get(1)).exists()  },
                { assert path(process.out.dbcandiamond_results.get(0).get(1)).exists()  }
            )
        }
    }

    test("dbcancazyme - cazyme annotation - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [[id: 'stub'],file('stub') ]
                input[1] = file('stub_db')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert path(process.out.cazyme_annotation.get(0).get(1)).exists() },
                { assert path(process.out.dbcanhmm_results.get(0).get(1)).exists() },
                { assert path(process.out.dbcansub_results.get(0).get(1)).exists()  },
                { assert path(process.out.dbcandiamond_results.get(0).get(1)).exists()  }


            )
        }
    }
}
