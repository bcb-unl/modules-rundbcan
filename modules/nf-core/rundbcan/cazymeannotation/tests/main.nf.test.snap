{
    "dbcancazyme - cazyme annotation - stub": {
        "content": [
            [
                "versions.yml:md5,8435140fdcf26095886f09384f61b273"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-21T10:06:38.698639"
    },
    "dbcancazyme - simplified": {
        "content": [
            [
                "versions.yml:md5,8435140fdcf26095886f09384f61b273"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-21T10:06:35.087564"
    }
}